File: LAB QUIZZES/CSE221_Quiz05/_A6C/_A6C_Testing.py
import os, sys, ast, time, random, threading, subprocess
# Compile and run: pypy _A6C_Testing.py
from collections import deque
def solve(N, x1, y1, x2, y2):
    '''
    N = int(input())
    x1, y1  = map(int, input().split())
    x2, y2  = map(int, input().split())
    '''
    moves = [(2, 1), (2, -1), (-2, 1), (-2, 1,), (-2, -1), (1, 2), (1, -2), (-1, 2), (-1, -2)]
    dist= [[-1]*(N+1) for _ in range(N+1)]
    que = deque()
    dist[x1][y1] = 0
    que.append((x1, y1))
    while que:
        #print(que)
        x, y = que.popleft()
        if (x, y) == (x2, y2):
            break
        for dx, dy in moves:
            nx, ny = x+dx, y+dy
            if 1 <= nx <= N and 1 <= ny <= N and dist[nx][ny] == -1:
                dist[nx][ny] = dist[x][y] + 1
                que.append((nx, ny))
    return(dist[x2][y2])


# stdin
# 4
# 1 1
# 3 4

# N = int(input())
# X1, Y1 = list(map(int, input().split()))
# X2, Y2 = list(map(int, input().split()))

# # hardcoded input
N = 3
X1 = 2
Y1 = 2
X2 = 1
Y2 = 3

start = time.time()
res = solve(N, X1, Y1, X2, Y2)
finish = time.time()
elapsed = finish - start

# # file output
# with open("OutputOfYourCode.txt", "w") as fout:
#     fout.write(str(res))

# stdout
print(res)

print("Execution time for the targeted one test is " + f"{elapsed:.9f}" + "s", flush=True)
os._exit(0)

File: LAB QUIZZES/CSE221_Quiz05/_A6C/_A6C_Testing.java
import java.util.*;
import java.io.*;

// Compile and run: javac _A6C_Testing.java && java _A6C_Testing
public class _A6C_Testing {
    public static class Pair<K, V> {
        public K key;
        public V value;

        public Pair(K key, V value) {
            this.key = key;
            this.value = value;
        }

        public String toString() {
            return "(" + key + ", " + value + ")";
        }
    }
    public static Integer solve(Integer N, Integer X1, Integer Y1, Integer X2, Integer Y2) {
        Integer res = 0;
        // Place your code here
        return res;
    }

    public static void main(String args[]) throws Exception {
        // stdin
        // 4
        // 1 1
        // 3 4
        // 
        // BufferedReader fin = new BufferedReader(new InputStreamReader(System.in));
        // String[] tempA = fin.readLine().split(" ");
        // int N = Integer.parseInt(tempA[0]);
        // tempA = fin.readLine().split(" ");
        // int X1 = Integer.parseInt(tempA[0]);
        // int Y1 = Integer.parseInt(tempA[1]);
        // tempA = fin.readLine().split(" ");
        // int X2 = Integer.parseInt(tempA[0]);
        // int Y2 = Integer.parseInt(tempA[1]);


        // // hardcoded input
        int N = 4;
        int X1 = 1, Y1 = 1;
        int X2 = 3, Y2 = 4;
        

        long start = System.nanoTime();
        Integer res = solve(N, X1, Y1, X2, Y2);
        long finish = System.nanoTime();
        long elapsed = finish - start;

        // // file output
        // BufferedWriter fout = new BufferedWriter(new
        // FileWriter("OutputOfYourCode.txt"));
        // fout.write(res.toString());
        // fout.close();

        // stdout
        System.out.println(res);

        System.out.printf("Execution time for the targeted one test is %.9fs\n", elapsed * 1e-9);
        System.exit(0);
    }
}
